lib/network-layer/src/Cardano/Wallet/Network/Light.hs

```diff
@@ -113,23 +113,39 @@ type LightBlocks m block addr txs =
     Either (NonEmpty block) (BlockSummary m addr txs)
 
 -- | Retrieve the 'ChainPoint' with the highest 'Slot'.
-latest :: [ChainPoint] -> ChainPoint
-latest [] = ChainPointAtGenesis
+latest :: [Read.ChainPoint] -> Read.ChainPoint
+latest [] = Read.GenesisPoint
 latest xs = maximumBy compareSlot xs
 
 -- | Retrieve the 'ChainPoint' with the second-highest 'Slot'.
-secondLatest :: [ChainPoint] -> ChainPoint
-secondLatest []  = ChainPointAtGenesis
-secondLatest [_] = ChainPointAtGenesis
+secondLatest :: [Read.ChainPoint] -> Read.ChainPoint
+secondLatest []  = Read.GenesisPoint
+secondLatest [_] = Read.GenesisPoint
 secondLatest xs  = head . tail $ sortBy (flip compareSlot) xs
 
+-- | Compare the slot numbers of two 'Read.ChainPoint's,
+-- but where the 'Read.GenesisPoint' comes before all other slot numbers.
+compareSlot :: Read.ChainPoint -> Read.ChainPoint -> Ordering
+compareSlot pt1 pt2 = compare (toOrdered pt1) (toOrdered pt2)
+  where
+    toOrdered :: Read.ChainPoint -> Integer
+    toOrdered Read.GenesisPoint = -1
```

Comment:
with `0` it would not work?

Reply:
Yes. The points `GenesisPoint` and `Read.BlockPoint (Read.SlotNo 0)` are different.